lib/my_api_client/request/pagination.rb

```diff
@@ -18,12 +18,19 @@ module Pagination
       # @param body [Hash, nil]
       #   Request body. You should not specify it when use GET method.
       # @return [Enumerator::Lazy]
-      #   Yields the pagination API response.
+      #   Yields the pagination API response if the block is given.
+      #   Whatever the block returns if the block is given.
+      # @yield
+      #   Process the response body with the given block.
+      # @yieldparam response [Sawyer::Response]
+      #   Response instance.
+      # @yieldreturn [<T>]
+      #   The block result.
       def pageable_get(pathname, paging:, headers: nil, query: nil)
         Enumerator.new do |y|
           response = call(:_request_with_relative_uri, :get, pathname, headers, query, nil)
           loop do
-            y << response.data
+            y << (block_given? ? yield(response) : response.data)
 
             next_uri = JsonPath.new(paging).first(response.body)
             break if next_uri.blank?
```

Comment:
@dai_okumura
I felt that the implementation of `#pageable_get` did not behave as expected.
This is how PR implementation works:

```rb
# with a block
instance.pageable_get('pages/1', paging: '$.next', headers: headers, query: query) do |result|
  result # => #<Enumerator::Lazy>
  result.each do |response|
    response # => #<Sawyer::Response>
  end
end

# without a block
instance.pageable_get('pages/1', paging: '$.next', headers: headers, query: query).each do |result|
  result # => #<Sawyer::Resource>
end
```

I want `#pageable_get` to always return `Enumerator::Lazy`, so I don't want the return value to change depending on the presence or absence of a block.

If supported, the following behavior would be desirable, but it would probably be difficult to implement:

```rb
instance.pageable_get('pages/1', paging: '$.next', headers: headers, query: query).each do |result|
  result # => #<Sawyer::Resource>
  result do |response|
    response # => #<Sawyer::Response>
  end
end
```

If you implement it as follows, you will get a similar behavior:

```rb
y << lambda do
  block_given? ? yield(response) : response.data
end
```

```rb
instance.pageable_get('pages/1', paging: '$.next', headers: headers, query: query).each do |result|
  result.call # => #<Sawyer::Resource>
  result.call do |response|
    response # => #<Sawyer::Response>
  end
end
```

However, it would be a destructive change, so I think it would be better to postpone supporting `#pageable_get`.

Reply:
@ryz310 
I would revert the changes for `#pget` because I think the changes for `#pget` is too much for the purpose.
This is fixed in 31b5d733edaaf565f2fe6deceea8a903b9057ec4 .
Thank you for review.

Note: Your first block of the code is not that I expected.

```diff
 # with a block
 instance.pageable_get('pages/1', paging: '$.next', headers: headers, query: query) 
 do |result|
-   result # => #<Sawyer::Resource>
+  # Here is the block for pageable_get
+  result # => #<Sawyer::Response>
+  # e.g. The user can parse the response: 
+  # JSON.Parse(result.body)['item']
-  result.each do |response|
-    response # => #<Sawyer::Response>
-  end
+end.each do |response|
+  response # => #<Sawyer::Response>
+  # e.g. The user can receive: '"item" in the response body JSON item'
 end
```